x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/results/add_result_logic.test.ts

```diff
@@ -0,0 +1,133 @@
+
+describe('AddResultLogic', () => {
+  const { mount } = new LogicMounter(AddResultLogic);
+  const { http } = mockHttpValues;
+  const { flashAPIErrors } = mockFlashMessageHelpers;
+
+  const MOCK_SEARCH_RESPONSE = {
+    results: [
+      { id: { raw: 'document-1' }, _meta: { id: 'document-1', engine: 'some-engine' } },
+      { id: { raw: 'document-2' }, _meta: { id: 'document-2', engine: 'some-engine' } },
+      { id: { raw: 'document-3' }, _meta: { id: 'document-3', engine: 'some-engine' } },
+    ],
+  };
+
+  const DEFAULT_VALUES = {
+    isFlyoutOpen: false,
+    dataLoading: false,
+    searchQuery: '',
+    searchResults: [],
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('has expected default values', () => {
+    mount();
+    expect(AddResultLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('openFlyout', () => {
+      it('sets isFlyoutOpen to true', () => {
+        mount({ isFlyoutOpen: false });
+
+        AddResultLogic.actions.openFlyout();
+
+        expect(AddResultLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          isFlyoutOpen: true,
+        });
+      });
+    });
+
+    describe('closeFlyout', () => {
+      it('sets isFlyoutOpen to false', () => {
+        mount({ isFlyoutOpen: true });
+
+        AddResultLogic.actions.closeFlyout();
+
+        expect(AddResultLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          isFlyoutOpen: false,
+        });
+      });
+    });
+
+    describe('search', () => {
+      it('sets searchQuery & dataLoading to true', () => {
+        mount({ searchQuery: '', dataLoading: false });
+
+        AddResultLogic.actions.search('hello world');
+
+        expect(AddResultLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchQuery: 'hello world',
+          dataLoading: true,
+        });
+      });
+    });
+
+    describe('onSearch', () => {
+      it('sets searchResults & dataLoading to false', () => {
+        mount({ searchResults: [], dataLoading: true });
+
+        AddResultLogic.actions.onSearch(MOCK_SEARCH_RESPONSE);
+
+        expect(AddResultLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          searchResults: MOCK_SEARCH_RESPONSE.results,
+          dataLoading: false,
+        });
+      });
+    });
+  });
+
+  describe('listeners', () => {
+    describe('search', () => {
+      beforeAll(() => jest.useFakeTimers());
+      afterAll(() => jest.useRealTimers());
+
+      it('should make a GET API call with a search query', async () => {
+        http.get.mockReturnValueOnce(Promise.resolve(MOCK_SEARCH_RESPONSE));
+        mount();
+        jest.spyOn(AddResultLogic.actions, 'onSearch');
+
+        AddResultLogic.actions.search('hello world');
+        jest.runAllTimers();
+        await nextTick();
+
+        expect(http.get).toHaveBeenCalledWith(
+          '/api/app_search/engines/some-engine/curation_search',
+          { query: { query: 'hello world' } }
```

Comment:
This made me double take initially but it's legit lol. The first `query` is the payload format our server wants, and the 2nd `query` is param key we use for searching